src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp

```diff
@@ -277,16 +287,32 @@ namespace OpenMS
     // initialization
     output.clear();
 
-    // naive cleavage sites
-    std::vector<Size> pep_positions = tokenize_(sequence.getString());
-    Size count = pep_positions.size();
-
     // disable max length filter by setting to maximum length
     if (max_length == 0)
     {
       max_length = sequence.size();
     }
 
+    // Unspecific cleavage:
+    // For unspecific cleavages every amino acid is a cutting position.
+    // And all substrings of legnth minx_size to max_size are generated.
+    if (enzyme_.getName() == "unspecific cleavage")
```

Comment:
can we replace this with an enum or some sort of expression which is not as error-prone as this name?